hazelcast/src/test/java/com/hazelcast/spi/hashslot/HashSlotArrayImplTest.java

```diff
@@ -0,0 +1,286 @@
+package com.hazelcast.spi.hashslot;
+
+import com.hazelcast.internal.memory.MemoryAccessor;
+import com.hazelcast.memory.HeapMemoryManager;
+import com.hazelcast.memory.MemoryManager;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.Random;
+
+import static com.hazelcast.memory.MemoryAllocator.NULL_ADDRESS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class HashSlotArrayImplTest {
+
+    private static final int VALUE_LENGTH = 32;
+
+    private final Random random = new Random();
+    private MemoryManager memMgr;
+    private MemoryAccessor mem;
+    private HashSlotArray hsa;
+
+    @Before
+    public void setUp() throws Exception {
+        memMgr = new HeapMemoryManager(32 << 20);
```

Comment:
I think, at least we should have for native memory allocator. 
We had its test before this PR on EE side and I think we should keep it.

In addition, for example, in the `HashSlotArrayBase`, even though I free the wrong memory address on dispose, this test won't fail unless we track and check the allocated addresses in the `HeapMemoryManager$MemoryAllocator` (since this memory allocator is just for testing, simply store the allocated memories in a `HashMap` and just check the disposed memory address in the `free` method). 

Also even though we forget freeing the allocated memory on dispose, there will be no problem in this test. But in fact this will cause memory leak and this should be tested on EE by simply checking native memory usage is `0` or not after dispose.

It is very easy to have native memory version of this test:
``` java
public class HashSlotArrayImplTest {
    
    ...

    abstract protected MemoryManager createMemoryManager(long size);

    @Before
    public void setUp() throws Exception {
        memMgr = createMemoryManager(32 << 20);
        ...
    }

    ...

}
```
``` java
@RunWith(EnterpriseSerialJUnitClassRunner.class)
@Category(QuickTest.class)
public class NativeMemoryHashSlotArrayImplTest extends HashSlotArrayImplTest {

    @Override
    protected MemoryManager createMemoryManager(long size) {
        return new StandardMemoryManager(...);
    }

}
```
And thats all. We have all tests on EE based on native memory.